webhooks/virtualmachine/v1alpha2/mutation/virtualmachine_mutator.go

```diff
@@ -295,6 +297,35 @@ func SetDefaultPowerState(
 	return false
 }
 
+// SetDefaultBiosUUID sets a default bios uuid for a new VM.
+// If CloudInit is the Bootstrap method, CloudInit InstanceID is also set to BiosUUID.
+// Return true if a default bios uuid was set, otherwise false.
+func SetDefaultBiosUUID(
+	ctx *context.WebhookRequestContext,
+	client client.Client,
+	vm *vmopv1.VirtualMachine) bool {
+
+	var wasMutated bool
+
+	if vm.Spec.BiosUUID == "" {
+		vm.Spec.BiosUUID = uuid.New().String()
+		wasMutated = true
+	}
+
+	if vm.Spec.Bootstrap != nil && vm.Spec.Bootstrap.CloudInit != nil {
+		if vm.Annotations == nil {
```

Comment:
Nit: we could move this inside the `if id != vm.Spec.BiosUUID` block since that's when we need to write to the map.

Reply:
Done